java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfig.java

```diff
@@ -338,4 +356,18 @@ public Map<Integer, Map<String, String>> getColumnMetadataByColumnIndex() {
   public RoundingMode getBigDecimalRoundingMode() {
     return bigDecimalRoundingMode;
   }
+
+  /**
+   * Interface for a function with 4 parameters.
+   *
+     * @param <A> param 1 type
+     * @param <B> param 2 type
+     * @param <C> param 3 type
+     * @param <D> param 4 type
+     * @param <R> return type
+   */
+  @FunctionalInterface
+  interface Function4Arity<A, B, C, D, R> {
```

Comment:
1. public?
2. Let's make this interface specific to the getter (so JdbcConsumerFactory or similar) so it's clear what it's for? (Also then it doesn't need to be generic.)

Reply:
done! Though I realized that most implementations will likely also need the config itself as one of the parameters... bit of a chicken and egg problem. You don't have the config yet when building the config -.-

I'm thinking there might be some way to do this but I'm not familiar enough with java to think of a good solution. I imagine some sort of wrapper or abstract class that injects the instance of the config when `apply` is called but I'm not exactly sure what this would look like.

Reply:
Ah, I forgot about that.

Can't you just make it one of the parameters? You're not going to invoke this while building the config, so it's fine to have it as a parameter.

Reply:
Oh right, lol not sure why I was overcomplicating it. FIxed